aws-api-request.js

```diff
@@ -9,6 +9,7 @@ function awsApiRequest(options, retryAttempt = 0) {
             accessKey = options.accessKey || awsApiRequest.accessKey || process.env.AWS_ACCESS_KEY_ID,
             secretKey = options.secretKey || awsApiRequest.secretKey || process.env.AWS_SECRET_ACCESS_KEY,
             sessionToken = options.sessionToken || awsApiRequest.sessionToken || process.env.AWS_SESSION_TOKEN,
+            maxBackoffRetries =  options.maxBackoffRetries || awsApiRequest.maxBackoffRetries || 10,
```

Comment:
This could then be simplified to:
```suggestion
            maxBackoffRetries =  options.maxBackoffRetries || awsApiRequest.maxBackoffRetries,
```
Or does this ruin something in the `options.accessKey`/`awsApiRequest.accessKey` relationship?

Reply:
This would effectively remove the fallback to awsApiRequest.maxBackoffRetries, entirely, from my eyes.

We want it to be the user option, then the internal option, then 10 if no one fills anything as a final catch.
Setting the default on the user option would mean the awsApiRequest value is never used. In effect, the internal API here can't set its own default values.

Thoughts?

Reply:
I see your point. Then let's get this shipped :)